src/app/shared/components/product-item/product-item.component.html

```diff
@@ -0,0 +1,63 @@
+<ng-container *ngIf="initShow; else short">
+  <div *ngIf="showFull" class="product-item" [ngClass]="{'out-of-stock': !product.availability.length}" (mouseover)="!product.availability.length ?hoverStateOut() : hoverStateIn()">
```

Comment:
logic in html detected

create getter isOutOfStock

and get rid of mouseover if-else check